providers/implementations/kdfs/hkdf.c

```diff
@@ -278,14 +278,15 @@ static int hkdf_common_set_ctx_params(KDF_HKDF *ctx, const OSSL_PARAM params[])
 static int kdf_hkdf_set_ctx_params(void *vctx, const OSSL_PARAM params[])
 {
     KDF_HKDF *ctx = vctx;
+    int seen_info = 0;
 
     if (params == NULL)
         return 1;
 
     if (!hkdf_common_set_ctx_params(ctx, params))
         return 0;
 
-    if (ossl_param_get1_concat_octet_string(params, OSSL_KDF_PARAM_INFO,
+    if (ossl_param_get1_concat_octet_string(params, OSSL_KDF_PARAM_INFO, &seen_info,
```

Comment:
Sorry...I'm still missing something here.

> For these, the param can be repeated and is concatenated. 

Where does the processing of that repetition happen? There is no loop inside `ossl_param_get1_concat_octet_string` or in this function?

> it is a "do this one" flag. It's set inside the function.

That much I understand. But it immediately goes out of scope and so is never used again for a subsequent instance of `OSSL_KDF_PARAM_INFO`???

Reply:
Here:

https://github.com/openssl/openssl/blob/2c500578fc68871eca7fabc7ee36b4027891671b/crypto/params.c#L1536-L1565


It becomes relevant once these are converted to use a loop-swtich structure which hasn't been done yet.
I expect these will be converted but the exact list hasn't been identified yet.

Reply:
Ok...thanks for pointing out the loop. But that still doesn't really explain the need for the `seen_info`. Currently it appears completely unused.

> It becomes relevant once these are converted to use a loop-swtich structure which hasn't been done yet.

In which case this so-far-unused change seems premature. Let's not include it now but when we actually need to use it. It appears quite separate to the rest of this PR.

Reply:
As soon as any of these are converted to the alternative:
```
for (p in params)
    switch (p->key)
```
this will be required because the latter params (with the same key) will double trigger.

I can convert.  I think they'd make the cut for desirability.